### ISSUE TITLE
`SceneTree.quit()` ignores given exit code with `--script`

### ISSUE THREAD

**Issue by razcore-rad** [godotengine/godot#88055]
### Tested versions

- Reproducible in: 4.2.1.stable.official.b09f793f5, 4.2.2.rc1.official.c7fb0645a

### System information

Ubuntu 23.10, Vulkan, NVIDIA GeForce RTX™ 2060 / NVIDIA GeForce RTX™ 2060

### Issue description

The following code:

```gdscript
extends SceneTree


func _init() -> void:
	quit(1)
```

saved as `dev.gd`, when ran with `godot --headless --script dev.gd` always returns with a `0` exit code:

```sh
❯ godot --headless --script dev.gd | complete
╭───────────┬──────────────────────────────────────────────────────────────────────╮
│ stdout    │ Godot Engine v4.2.2.rc1.official.c7fb0645a - https://godotengine.org │
│           │                                                                      │
│           │                                                                      │
│ exit_code │ 0                                                                    │
╰───────────┴──────────────────────────────────────────────────────────────────────╯
```

### Steps to reproduce

Check issue description.

### Minimal reproduction project (MRP)

not applicable

**Comment by akien-mga** [comment]
I can reproduce it with `--script`, but not when running an actual project.

**Comment by LeoBelda** [comment]
I can't reproduce it on Windows.
It seems to be caused by a call to `OS::set_exit_code` in the Linux `main`:
https://github.com/godotengine/godot/blob/5f05e2b9b1a3fedcdd7ecb2ab976a2687fd6f19a/platform/linuxbsd/godot_linuxbsd.cpp#L84-L87
Copying this line into Windows `widechar_main` allowed me to reproduce the issue: the exit code set by the script during the execution of `Main::start` gets overridden.

**Comment by akien-mga** [comment]
Fixed by #89229.
